src/Storages/MergeTree/ActiveDataPartSet.cpp

```diff
@@ -56,15 +58,18 @@ bool ActiveDataPartSet::add(const String & name, Strings * out_replaced_parts)
     /// Let's go to the right.
     while (it != part_info_to_name.end() && part_info.contains(it->first))
     {
-        if (part_info == it->first)
-            throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected duplicate part {}. It is a bug.", name);
+        assert(part_info != it->first);
```

Comment:
Logging was lost?

Reply:
It was redundant, see https://github.com/ClickHouse/ClickHouse/pull/24723#issuecomment-850649761